plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -527,12 +527,18 @@ public static LongWriteFunction timestampWriteFunction(TimestampType timestampTy
         return (statement, index, value) -> statement.setObject(index, fromTrinoTimestamp(value));
     }
 
-    public static ObjectWriteFunction longTimestampWriteFunction(TimestampType timestampType)
+    public static ObjectWriteFunction longTimestampWriteFunction(TimestampType timestampType, int roundToPrecision)
     {
         checkArgument(timestampType.getPrecision() > TimestampType.MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
+        checkArgument(
+                6 <= roundToPrecision && roundToPrecision <= 9 && roundToPrecision <= timestampType.getPrecision(),
```

Comment:
Shouldn't this be `6 < roundToPrecision`? Equals to 6 shouldn't reach long-timestamp code.

Reply:
i can CTAS timestamp(12) to precision 6 in the remote thingy, so i want allow 6.

actually, i could map it to timestamp(3) or even timestamp(0), just the code would need to be more complex, and doesn't need to be

Reply:
This 6 is arbitrary then. And the only reason the `checkArgument` doens't throw is because there is no JDBC connector whose max supported timestamp precision is < 6.

Do we have tests which cause rounding for `timestamp(p < 6)`?

Reply:
> This 6 is arbitrary then.

not quite. it's a reflection on the implementation. I believe the code works correctly for 6, but definitely is incorrect for 5.

now, i could fix the code to work correctly for 5 or less, but that would be useless (dead code), since only 6 or more is needed

> Do we have tests which cause rounding for timestamp(p < 6) so that if in future we have connector which only supports milliseconds 

No. Connectors 

- support timestamp(p) correctly (with full test coverage), OR
- support only timestamp(3) (so no issue here)
- don't support timestamp writes at all


I would expect that, if the above changes, it's appropriately test-covered, and the code adjusted as needed.

Reply:
Thanks for taking the time to explain. I see that the type mapping tests do cover it.